R/dataset.R

```diff
@@ -88,6 +88,42 @@ setMethod("notes<-", "CrunchDataset", function (x, value) {
     invisible(setEntitySlot(x, "notes", value))
 })
 
+
+#' Name, alias, and description for Crunch objects
+#'
+#' @param x a Dataset
+#' @param value For the setters, a single Variable to use as the primary key.
+#' @return Getters return the Variable object that is used as the primary key (`NULL` if there is no primary key); setters
+#' return \code{x} duly modified.
+#' @name pk
+#' @aliases pk pk<-
+NULL 
+
+#' @rdname pk
+#' @export
+setMethod("pk", "CrunchDataset", function (x)  {
+    pk <- ShojiEntity(crGET(shojiURL(x, "fragments", "pk")))$pk
+    if (length(pk)) {
+        return(x[[pk[[1]]]])
+    } else {
+        return(NULL)
+    }
+})
+#' @rdname pk
+#' @export
+setMethod("pk<-", "CrunchDataset", function (x, value) {
+    if (is.null(value)) {
+        crDELETE(shojiURL(x, "fragments", "pk"))
+    } else {
+        payload <- toJSON(structure(list(structure(list(self(value)))),
```

Comment:
You may be able to express this more concisely as `list(pk=I(self(value)))` because `I` ensures that you get an array in JSON even if the R vector is length 1.